File: src/main/java/handler/DraftKingsHandler.java
package handler;

import api.ApiCaller;
import api.DataCollector;
import collect.dfs.DraftKings;

import java.util.List;
import java.util.Map;

public class DraftKingsHandler {
    DataCollector dataCollector = new DataCollector(new ApiCaller());
    DraftKings draftKings = new DraftKings(dataCollector);

    public List<Map<String, Object>> handleRequest(Map<String, String> input) {
        String sport = input.get("sport");
        return draftKings.getAllContestData(sport);
    }
}

File: src/main/kotlin/handler/DraftKingsHandler.kt
package handler

import collect.dfs.DraftKingsContests

class DraftKingsHandler {
    fun handleRequest(input: Map<String, String>): List<Map<String, Any>> {
        val sport = input.getValue("sport")
        return getDraftKingsContestData(sport)
    }

    fun getDraftKingsContestData(sport: String): List<Map<String, Any>> {
        return DraftKingsContests().getDraftKingsContestData(sport)
    }
}